lxd/instance/drivers/driver_lxc.go

```diff
@@ -7112,14 +7106,16 @@ func (d *lxc) getFSStats() (*metrics.MetricSet, error) {
 			}
 
 			// Get the volume.
-			vol := pool.GetVolume(volType, storageDrivers.ContentTypeFS, volName, nil)
-			if !pool.Driver().HasVolume(vol) {
-				continue
+			tmpPool, err := storagePools.NewTemporary(d.state, &api.StoragePool{Name: dev["pool"], Driver: "dir"})
```

Comment:
Ive just seen this and I would prefer this use a less 'workaround' approach to get the volume mount path (as NewTemporary is for use with lxd import and hard coding driver type is not ideal).

Instead try using

volStorageName := project.Instance(inst.Project(), inst.Name())
mountPath := storageDrivers.GetVolumeMountPath(...)

Reply:
Fixed